File: AddressOwnershipTool.Web/ClientApp/src/app/models/claim-group.ts
export interface ClaimGroup {
  destination: string;
  origin: string;
  numberOfClaimedAddresses: number;
  totalAmountToTransfer: number;
  originalTotalBalance: number;
  claims: Claim[];
  type: string;
}

export interface ClaimGroupResponse {
  result: ClaimGroup[],
  message: string
}

export interface Claim {
  destination: string;
  origin: string;
  balance: number;
  originNetwork: string;
  type: string;
}

export interface SwappedTx {
  destination: string;
  origin: string;
  amount: number;
  txHash: string;
  type: string;
}

export interface SwapRequest {
  destination: string;
  origin: string;
  amount: number;
  txHash: string;
  path: string;
  type: string;
}

File: AddressOwnershipTool/Commands/Load/ClaimGroup.cs
namespace AddressOwnershipTool.Commands.Load;

public class ClaimGroup
{
    public string Destination { get; set; }

    public string Origin { get; set; }

    public string Type { get; set; }

    public int NumberOfClaimedAddresses => this.Claims.Count();

    public decimal TotalAmountToTransfer => this.OriginalTotalBalance * 10M;

    public decimal OriginalTotalBalance => this.Claims.Sum(c => c.Balance);

    public List<Claim> Claims { get; set; } = new List<Claim>();
}

public class Claim
{
    public string Destination { get; set; }

    public decimal Balance { get; set; }

    public string OriginNetwork { get; set; }

    public string Origin { get; set; }

    public string Type { get; set; }
}
